driver-sync/src/test/functional/com/mongodb/client/ChangeStreamProseTest.java

```diff
@@ -59,6 +61,34 @@ public void setUp() {
         collection.insertOne(Document.parse("{ _id : 0 }"));
     }
 
+    class ChangeStreamWatchThread extends Thread {
+        @Override
+        public void run() {
+            try {
+                for (final ChangeStreamDocument<Document> doc : collection.watch()) {
```

Comment:
To make sure we're interrupting resumability and not the initial aggregate, let's pass in the MongoCursor from calling `collection.watch().cursor()` to this class's constructor instead of creating it in the thread.  That way we're sure we're interrupting a call to `hasNext()` and not the call to `cursor()`.

Reply:
And if we do that, we probably don't need the `Thread.sleep()` call either, which will speed up the tests.

Reply:
Done